src/ir/ty.rs

```diff
@@ -568,7 +568,8 @@ impl Type {
         let canonical_ty = ty.canonical_type();
         let kind = match ty.kind() {
             CXType_Unexposed if *ty != canonical_ty &&
-                                canonical_ty.kind() != CXType_Invalid => {
+                                canonical_ty.kind() != CXType_Invalid &&
+                                !canonical_ty.spelling().contains("type-parameter") => {
```

Comment:
Added a comment, since I didn't find a better name (`type_is_not_canonical` may be a bit misleading).

Also commented with another plausible solution, worth trying in a followup maybe.